### PR TITLE
[Triangle] fix crash on windows + new clean upstream source

### PR DESCRIPTION
**Pull request by j-fu**
* Fix issue https://github.com/JuliaGeometry/Triangulate.jl/issues/15 by replacing `unsigned long` with `unsigned long long` on x86_64_w64_mingw32

* Upstream source with all patches is now https://github.com/JuliaGeometry/Triangle

* Synchronized version number with upstream source

### CODE REVIEW

**Comment by j-fu** [comment]
See [this discussion](https://github.com/JuliaGeometry/Triangle/issues/1) on tracking upstream source.

File: T/Triangle/build_tarballs.jl [hunk 1]
```diff
@@ -1,46 +1,36 @@
...
-$CC -Itriangle  -Wno-int-to-pointer-cast  -Wno-pointer-to-int-cast -DREAL=double -DTRILIBRARY -O3 -fPIC -DNDEBUG -DNO_TIMER -DEXTERNAL_TEST $LDFLAGS --shared -o "${libdir}/libtriangle.${dlext}" triangle_patched.c triwrap.c
-
-install_license triangle/README
+GENERICFLAGS=(-DREAL=double  -DTRILIBRARY  -DNDEBUG -DNO_TIMER -DEXTERNAL_TEST)
+
+ARCHFLAGS=(-DULONG="unsigned long" -fPIC)
+
+if [[ "${target}" == x86_64-w64-mingw32 ]]; then
+    ARCHFLAGS=(-DULONG="unsigned long long")
+fi
```
Comment (giordano) [inline T/Triangle/build_tarballs.jl:23]: I think
```suggestion
if [[ "${target}" == x86_64-w64-mingw32 ]]; then
    ARCHFLAGS=(-DULONG="unsigned long long")
else
    ARCHFLAGS=(-DULONG="unsigned long" -fPIC)
fi
```
read slightly clearer (at the beginning I thought the `if` was adding new elements to the original `ARCHFLAGS`, instead they're just two different settings).

File: T/Triangle/build_tarballs.jl [hunk 2]
```diff
@@ -1,46 +1,36 @@
...
+
+$CC "${ARCHFLAGS[@]}"  "${GENERICFLAGS[@]}" $LDFLAGS -O3  --shared -o "${libdir}/libtriangle.${dlext}" triangle.c triwrapjulia.c
```
Comment (giordano) [inline T/Triangle/build_tarballs.jl:25]: There is no `LDFLAGS`
```suggestion
$CC "${ARCHFLAGS[@]}"  "${GENERICFLAGS[@]}" -O3  --shared -o "${libdir}/libtriangle.${dlext}" triangle.c triwrapjulia.c
```

File: T/Triangle/build_tarballs.jl [hunk 3]
```diff
@@ -1,46 +1,36 @@
...
+
+install_license README
 """
 
 platforms = supported_platforms(; experimental=true)
```
Comment (giordano) [inline T/Triangle/build_tarballs.jl:30]: This PR didn't change this line, but while we're at it we can remove some craft (`experimental=true` has no effect nowadays):
```suggestion
platforms = supported_platforms()
```

**Pull request merged**